### PR TITLE
Authorize user

### PR DESCRIPTION
**Pull request by itsmohmans**

### CODE REVIEW

File: components/login/Form.vue [hunk 1]
```diff
@@ -105,30 +66,17 @@ const storeMemberData = () => {
...
-    })
-    .catch((error) => {
-      console.log(error)
-    })
-    .finally(() => {
-      userAuth()
     })
+  })
+  // userAuth()
+  const {data: member} = await useFetch(`${config.public.COMMUNITY_API_URL}/auth/`)
```
Comment (itsmohmans) [inline components/login/Form.vue:79]: To fix: session cookies are not sent with this request. See https://stackoverflow.com/a/5422323

File: middleware/auth.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default defineNuxtRouteMiddleware(async () => {
```
Comment (thamudi) [inline middleware/auth.js:1]: Add the route members/[id] to the list that checks if the user is auth or not. 
For the profile page, if the user is not authenticated just show the regular page without the edit page options.

File: components/login/Form.vue [hunk 3]
```diff
@@ -25,19 +25,39 @@
...
+    method: "POST",
+    body: JSON.stringify({
+      "email": form.email,
+      "password": form.password
+    }),
+    onResponse({response}) {
+      if(response.ok) {
+        localStorage.setItem("member", JSON.stringify(response._data))
+      }
+    },
```
Comment (thamudi) [inline components/login/Form.vue:55]: Redirect the auth member to the home page.

File: components/UserWidget.vue [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+<script setup>
+const togglePopup = () => {
+  state.isOpen = !state.isOpen
+}
+
+const signOut = async () => {
+  togglePopup()
+  await useFetch('/api/logout')
+  localStorage.clear()
+  location.reload()
```
Comment (thamudi) [inline components/UserWidget.vue:50]: instead of reload, redirect the user to the homepage

File: server/api/forgot-password/send-email.post.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export default defineEventHandler(async (event) => {
+  const config = useRuntimeConfig()
+  const body = await useBody(event)
+  const options = {
+    method: 'GET',
```
Comment (thamudi) [inline server/api/forgot-password/send-email.post.js:5]: change this to a POST instead once the endpoint on the API is updated

File: pages/members/[id].vue [hunk 6]
```diff
@@ -30,7 +30,9 @@
 const config = useRuntimeConfig()
 const route = useRoute()
 const user_id = route.params.id
-
+onMounted(() => {
```
Comment (thamudi) [inline pages/members/[id].vue:33]: Double check this file, it should already in the development, try rebasing

**Review by thamudi — CHANGES_REQUESTED** [review]
Please check the comments and update the branch accordingly

**Review by thamudi — APPROVED** [review]
Reviewed and approved

**Pull request merged**